### PR TITLE
Fixes #1605: Customizable metadata key on encryption key providers

### LINKED ISSUE #1605: encryption: changing key provider name breaks it
### OpenTofu Version

```shell
OpenTofu v1.7.0
```

### OpenTofu Configuration Files

Changing from

```hcl
    key_provider "aws_kms" "aes_256" {
      kms_key_id = "mrk-1"
      region     = "us-east-1"
      key_spec   = "AES_256"
    }
```

to

Changing from

```hcl
    key_provider "aws_kms" "hello" {
      kms_key_id = "mrk-1"
      region     = "us-east-1"
      key_spec   = "AES_256"
    }
```

Breaks the config and will result in `Error: error loading state: decryption failed for all provided methods: attempted decryption failed for state: no decryption key available`

### Debug Output

.

### Expected Behavior

This is unexpected for terraform config, since when does resource naming change anything?

### Actual Behavior

.

### Steps to Reproduce

tofu apply

### Additional Context

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ghost**
This PR adds the ability to customize the metadata key for encryption key providers by specifying the `encrypted_metadata_alias` field:

```hcl2
terraform {
  encryption {
    key_provider "pbkdf2" "basic" {
      encrypted_metadata_alias = "foo"
      passphrase             = "Hello world! 123"
    }
    method "aes_gcm" "example" {
      keys = key_provider.pbkdf2.basic
    }
    state {
      method = method.aes_gcm.example
    }
  }
}
```

Resolves #1605

## Target Release

1.9.0

## Checklist

- [x] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [x] I have not used an AI coding assistant to create this PR.
- [x] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.
- [x] I (and other contributors to this PR) have not looked at the Terraform source code while implementing this PR.

### Go checklist

 

- [x] I have run golangci-lint on my change and receive no errors relevant to my code.
- [x] I have run existing tests to ensure my code doesn't break anything.
- [x] I have added tests for all relevant use cases of my code, and those tests are passing.
- [x] I have only exported functions, variables and structs that should be used from other packages.
- [x] I have added meaningful comments to all exported functions, variables, and structs.

### Website/documentation checklist

- [x] I have locally started the website as [described here](https://github.com/opentofu/opentofu/blob/main/website/README.md) and checked my changes.

### CODE REVIEW

File: website/docs/language/state/encryption.mdx [hunk 1]
```diff
@@ -148,22 +147,24 @@ The PBKDF2 key provider allows you to use a long passphrase as to generate a key
...
+| key_length              | Number of bytes to generate as a key.                                                                                                                   | 1         | 32                                 |
+| iterations              | Number of iterations. See [this document](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2) for recommendations. | 200.000   | 600.000                            |
+| salt_length             | Length of the salt for the key derivation.                                                                                                              | 1         | 32                                 |
+| hash_function           | Specify either `sha256` or `sha512` to use as a hash function. `sha1` is not supported.                                                                 | N/A       | sha512                             |
+| encrypted_metadata_key  | Optional identifier to store metadata in the encrypted state/plan files under. Specify this to freely change the name of a key provider.                | -         | derived from the key provider name |
```
Comment (cam72cam) [inline website/docs/language/state/encryption.mdx:157]: suggestion "to freely change" -> "to allow changing"?

**Review by cam72cam — APPROVED** [review]

**Review by ollevche — APPROVED** [review]
Looks good overall, left a few nit comments.

**Comment by cam72cam** [comment]
Documenting that we had a conversation about potentially renaming the field for clarity

**Review by ollevche — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ UPGRADE NOTES:
 NEW FEATURES:
 
 ENHANCEMENTS:
+* State encryption key providers now support customizing the metadata key via `encrypted_metadata_key` ([#1605](https://github.com/opentofu/opentofu/issues/1605))
```
Comment (cam72cam) [inline CHANGELOG.md:10]: encrypted_metadata_alias

File: website/docs/language/state/examples/encryption/terraform_remote_state_full_a.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
+terraform {
+  encryption {
+    key_provider "pbkdf2" "mykey" {
+      passphrase             = "OpenTofu has encryption"
+      # Note the fixed encryption_metadata_key here:
```
Comment (cam72cam) [inline website/docs/language/state/examples/encryption/terraform_remote_state_full_a.tf:5]: encrypted_metadata_alias

File: website/docs/language/state/examples/encryption/terraform_remote_state_full_b.tf [hunk 4]
```diff
@@ -0,0 +1,32 @@
+terraform {
+  encryption {
+    # Note that the name of the key here is different:
+    key_provider "pbkdf2" "mykeyrenamed" {
+      passphrase             = "OpenTofu has encryption"
+      # Note the fixed encryption_metadata_key here:
```
Comment (cam72cam) [inline website/docs/language/state/examples/encryption/terraform_remote_state_full_b.tf:6]: encrypted_metadata_alias

**Review by cam72cam — APPROVED** [review]

**Review by ollevche — APPROVED** [review]

**Pull request merged**